src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj

```diff
@@ -11,8 +11,14 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <PropertyGroup>
```

Comment:
@eerhardt \ @MichaelSimons wanted to get your thoughts on this, as an enhancement to https://github.com/dotnet/sdk/pull/22072/files. Users are often affected by this when running a fairly new SDK using MSBuild where the compiler hasn't been updated as yet.

Reply:
I think this is a bigger issue / question for the SDK and Roslyn teams.  cc @jaredpar @chsienki @dsplaisted 

So the repro steps here are:

1. Install VS 2022, but not the latest version
2. Install .NET Core SDK 6.0.200

and now we don't have a coherent SDK + VS?  There are components in the 6.0.200 SDK that reference Roslyn `4.1`, but the VS 2022 installed only has Roslyn `4.0`?

Is that all true?

IMO - we shouldn't be using different version when building from source vs. building for Microsoft distribution. That's a recipe for disaster.

Reply:
Interestingly, this appears to be the path the ApiCompatibility tool has taken - https://github.com/dotnet/sdk/pull/23321/files

Reply:
Agreed - I wasn't excited about that change either 😄. The difference with the ApiCompatibility tool is that it also ships OOB from the SDK into it's own NuGet package:

https://www.nuget.org/packages/Microsoft.DotNet.Compatibility

So someone could be using an older SDK, and PackageReference the newer version of that NuGet package.

Correct me if I'm wrong, but I believe that the Razor Source Generator only ships as part of the SDK, right? Shouldn't we have a guaranteed version of Roslyn that the SDK supports?

Reply:
> Correct me if I'm wrong, but I believe that the Razor Source Generator only ships as part of the SDK, right? 

That is right.

> Shouldn't we have a guaranteed version of Roslyn that the SDK supports?

👍🏽

Reply:
Pinning the compiler means that you're not getting bug fixes, features, etc ... 

It's less important for a tool like API Compat which is using us for a limited use case. Mostly to grab a semantic model to inspect the code. Much more important for components that are actually participating in mainline compilation. 

In this case though this is not pinning the compiler, it's pinning the Compiler API version Razor is compiling against. At runtime we're going to use the MS.CA that ships with the compiler. Still though this feels very unnecessary.

Reply:
@jaredpar do you have suggestions on how we should proceed here? Users can unblock themselves by updating their VS / pinning the SDK, but both of these seem like crummy option.

Would we be open to taking this PR for 6.0.x and consider something more lasting for 7.0?

Reply:
Is this the only analyzer / source generator that actually ships in box with the SDK? I know that the roslyn-analyzer pin their version way back to something like `3.3`. 

We could follow that path here, and then for source-build, we need to put the `ref assemblies` for the version you pin in the "SBRP".

https://github.com/dotnet/roslyn-analyzers/blob/eb9a7a5d2e4dcbfd4db89c2ee48e79bca107484c/eng/Versions.props#L45-L57

Reply:
> Is this the only analyzer / source generator that actually ships in box with the SDK?

ASP.NET Core builds some analyzers that ship as part of the WebSDK. Those are currently pinned to build against 3.3:

https://github.com/dotnet/aspnetcore/blob/main/eng/Versions.props#L198-L199

Reply:
Opened https://github.com/dotnet/source-build-reference-packages/pull/357 to add the required bits to SBRP so we can target the same version in source build.